tests/cmdline/commands/test_presto.py

```diff
@@ -0,0 +1,31 @@
+"""Tests for ``verdi presto``."""
+
+import pytest
+from aiida.cmdline.commands.cmd_presto import default_profile_name, verdi_presto
+from aiida.manage.configuration.config import Config
+
+
+@pytest.mark.parametrize(
+    'profile_names, expected',
+    (
+        ([], 'presto'),
+        (['main', 'sqlite'], 'presto'),
+        (['presto'], 'presto-1'),
+        (['presto', 'presto-5', 'presto-2'], 'presto-6'),
+        (['presto', 'main', 'presto-2', 'sqlite'], 'presto-3'),
+    ),
+)
+def test_default_profile_name(monkeypatch, profile_names, expected):
+    """Test the dynamic default profile function."""
+
+    def get_profile_names(self):
+        return profile_names
+
+    monkeypatch.setattr(Config, 'profile_names', property(get_profile_names))
+    assert default_profile_name() == expected
+
+
+def test_presto(run_cli_command):
+    """Test the ``verdi presto``."""
+    result = run_cli_command(verdi_presto, use_subprocess=False)
```

Comment:
I am not familiar with CLI tests, are there any downsides of using `use_subprocess=False`?

Reply:
This is necessary if the test relies on changes that are made in memory but the CLI command would read things from disk. But actually here it is not necessary. I improved the test to use the new `aiida_config_tmp` fixture which creates an entire temporary config instance and then run the CLI command normally. This way it fully mimics the actual usage.

Reply:
Awesome, thanks for the explanation and changes!